google/cloud/bigquery/model.py

```diff
@@ -66,140 +61,122 @@ def __init__(self, model_ref):
-        is inherited from the dataset.
+    def location(self) -> str:
+        """The geographic location where the model resides.
+
+        This value is inherited from the dataset.
 
         Read-only.
         """
-        return self._proto.location
+        return self._properties.get("location")
 
     @property
-    def etag(self):
-        """str: ETag for the model resource (:data:`None` until
-        set from the server).
+    def etag(self) -> str:
+        """ETag for the model resource (:data:`None` until set from the server).
 
         Read-only.
         """
-        return self._proto.etag
+        return self._properties.get("etag")
 
     @property
-    def created(self):
-        """Union[datetime.datetime, None]: Datetime at which the model was
-        created (:data:`None` until set from the server).
+    def created(self) -> Optional[datetime.datetime]:
+        """Datetime at which the model was created (:data:`None` until set from the server).
 
         Read-only.
         """
-        value = self._proto.creation_time
-        if value is not None and value != 0:
+        value = self._properties.get("creationTime")
+        if value is not None:
             # value will be in milliseconds.
             return google.cloud._helpers._datetime_from_microseconds(
                 1000.0 * float(value)
             )
 
     @property
-    def modified(self):
-        """Union[datetime.datetime, None]: Datetime at which the model was last
-        modified (:data:`None` until set from the server).
+    def modified(self) -> Optional[datetime.datetime]:
+        """Datetime at which the model was last modified (:data:`None` until set from the server).
 
         Read-only.
         """
-        value = self._proto.last_modified_time
-        if value is not None and value != 0:
+        value = value = self._properties.get("lastModifiedTime")
+        if value is not None:
             # value will be in milliseconds.
             return google.cloud._helpers._datetime_from_microseconds(
                 1000.0 * float(value)
             )
 
     @property
-    def model_type(self):
-        """google.cloud.bigquery_v2.types.Model.ModelType: Type of the
-        model resource.
+    def model_type(self) -> str:
+        """Type of the model resource.
 
         Read-only.
-
-        The value is one of elements of the
-        :class:`~google.cloud.bigquery_v2.types.Model.ModelType`
-        enumeration.
         """
-        return self._proto.model_type
+        return self._properties.get("modelType", "MODEL_TYPE_UNSPECIFIED")
 
     @property
-    def training_runs(self):
-        """Sequence[google.cloud.bigquery_v2.types.Model.TrainingRun]: Information
-        for all training runs in increasing order of start time.
+    def training_runs(self) -> Sequence[Dict[str, Any]]:
+        """Information for all training runs in increasing order of start time.
```

Comment:
Let's make sure we link to the REST API docs.

```suggestion
        """Information for all training runs in increasing order of start time.

        Dictionaries are in REST API format. See:
        https://cloud.google.com/bigquery/docs/reference/rest/v2/models#trainingrun
```